tools/shared.py

```diff
@@ -931,8 +935,10 @@ def get_llvm_target():
 
 if LLVM_TARGET == WASM_TARGET:
   # wasm target does not automatically define emscripten stuff, so do it here.
-  COMPILER_OPTS = COMPILER_OPTS + ['-DEMSCRIPTEN',
-                                   '-D__EMSCRIPTEN__']
+  COMPILER_OPTS = COMPILER_OPTS + ['-D__EMSCRIPTEN__']
```

Comment:
Yes, I see, the hacks there need to only add it in some cases.

I suppose we could have a `add_emscripten_define` method somewhere, which returns the one define or both depending on if we are in strict mode. How does that sound?